discovery/http/http.go

```diff
@@ -154,7 +155,7 @@ func (d *Discovery) refresh(ctx context.Context) ([]*targetgroup.Group, error) {
 		return nil, errors.Errorf("server returned HTTP status %s", resp.Status)
 	}
 
-	if !matchContentType.MatchString(resp.Header.Get("Content-Type")) {
+	if !matchContentType.MatchString(strings.TrimSpace(resp.Header.Get("Content-Type"))) {
```

Comment:
> Leading and trailing whitespace in raw field values is removed upon
   field parsing (Section 3.2.4 of [RFC7230]).

(https://datatracker.ietf.org/doc/html/rfc7231#section-8.3.1)